src/libcommon/utility/utility.cpp

```diff
@@ -728,19 +728,11 @@ bool CommonUtility::isVersionLower(const std::string &currentVersion, const std:
     extractIntFromStrVersion(targetVersion, targetTabVersion);
 
     if (currTabVersion.size() != targetTabVersion.size()) {
-        // Should not happen
-        return false;
-    }
-
-    for (size_t i = 0; i < targetTabVersion.size(); i++) {
-        if (currTabVersion[i] > targetTabVersion[i]) {
-            return false;
-        } else if (currTabVersion[i] < targetTabVersion[i]) {
-            return true;
-        }
+        return false; // Should not happen
     }
 
-    return false;
+    return std::lexicographical_compare(currTabVersion.cbegin(), currTabVersion.cend(), targetTabVersion.cbegin(),
+                                        targetTabVersion.cend());
```

Comment:
```suggestion
    return std::ranges::lexicographical_compare(currTabVersion, targetTabVersion);
```

Reply:
Fantastic! Changed to the concise call in the next commit.

Reply:
Done.